clients/feeder/feeder.go

```diff
@@ -36,6 +36,8 @@ type Client struct {
 	userAgent  string
 	apiKey     string
 	listener   EventListener
+	timeouts   TimeoutsList
+	curTimeout int
```

Comment:
It looks to me that is the job of `TimeoutList` to know what the `curTimeout` is, meaning that `TimeoutsList` should be the one holding the index and not `Client`

Reply:
Moved to the `Timeouts` struct